### PR TITLE
updates #86 View Challenges Button to Lesson

### PR DESCRIPTION
**Pull request by SahilKalra98**

### CODE REVIEW

File: pages/curriculum/[lesson].tsx [hunk 1]
```diff
@@ -1,29 +1,60 @@
...
+    const lessonId = router.query.lesson
+    const lesson: Lesson = data.lessons.find(
+      (lesson: Lesson) => lesson.id.toString() === lessonId
+    )
+    return (
+      <div>
+        <Layout>
+          <div className="row mt-4">
+            <LessonTitleCard
+              lessonCoverUrl={`js-${lesson.order}-cover.svg`}
```
Comment (SahilKalra98) [inline pages/curriculum/[lesson].tsx:42]: Todo: Need to get correct image for Lesson. (Sort)

File: components/ChallengeMaterial.tsx [hunk 2]
```diff
@@ -35,20 +44,35 @@ export const ChallengeQuestionCard: React.FC<ChallengeQuestionCardProps> = props
   return <div className="card shadow-sm border-0">{content}</div>
 }
 
-const ChallengeMaterial: React.FC = () => {
-  return (
-    <div className="row">
-      <div className="col-4">
-        <ChallengeTitleCard challengeNum={1} title="Less Than or Equal to 5" />
+const ChallengeMaterial: React.FC<ChallengeMaterialProps> = props => {
+  if (props.challenges) {
```
Comment (songz) [inline components/ChallengeMaterial.tsx:48]: Avoid putting large code blocks inside an if. 
```
if (!props.challenges) {
  return <h1>Na Cha....
}
const sortedCha....
```

File: graphql/queries.ts [hunk 3]
```diff
@@ -30,3 +30,21 @@ export const GET_LESSONS = gql`
     }
   }
 `
+
+export const GET_CHALLENGES = gql`
+  query Lessons {
```
Comment (songz) [inline graphql/queries.ts:35]: do we need this

File: pages/curriculum/[lesson].tsx [hunk 4]
```diff
@@ -1,29 +1,60 @@
...
-        </div>
-      </Layout>
-    </div>
-  )
+  const router = useRouter()
+  const { loading, data } = useQuery(GET_CHALLENGES)
+  if (loading) {
+    return <h1>Loading</h1>
+  }
+  if (data) {
```
Comment (songz) [inline pages/curriculum/[lesson].tsx:32]:
```
if (!data) {
  return....
}
const lessonId = ....
```

You don't want big chunks of code in if code block.

File: pages/curriculum/[lesson].tsx [hunk 5]
```diff
@@ -1,25 +1,53 @@
...
+  if (!data) {
+    return <h1>...</h1>
+  }
+  const { lessons }: { lessons: Lesson[] } = data
+  const sortedLessons: Lesson[] = lessons.sort((a, b) => a.order - b.order)
+  const lessonId = router.query.lesson
+  const lessonIdx = sortedLessons.findIndex(
+    lesson => lessonId === lesson.id.toString()
+  )
+  const currentLesson = sortedLessons[lessonIdx]
```
Comment (songz) [inline pages/curriculum/[lesson].tsx:35]:
```
const currentLesson = lessons.find( lesson => router.query.lesson === lesson.id.toString())
```

File: pages/curriculum/[lesson].tsx [hunk 6]
```diff
@@ -1,25 +1,53 @@
...
   return (
     <div>
       <Layout>
         <div className="row mt-4">
           <LessonTitleCard
-            lessonCoverUrl="js-0-cover.svg"
-            lessonUrl="https://www.notion.so/JS-0-Foundations-a43ca620e54945b2b620bcda5f3cf672"
-            lessonTitle="Foundations"
+            lessonCoverUrl={`js-${lessonIdx}-cover.svg`}
```
Comment (songz) [inline pages/curriculum/[lesson].tsx:41]: `currentLesson.order`

File: pages/curriculum/[lesson].tsx [hunk 7]
```diff
@@ -1,25 +1,53 @@
...
+import { Challenge } from '../../@types/challenge'
+
+type Lesson = {
+  id: number
+  title: string
+  description: string
+  order: number
+  challenges: Challenge[]
+  docUrl: string
+}
```
Comment (songz) [inline pages/curriculum/[lesson].tsx:18]: Doesn't curriculum page also have a Lesson type already? Could we update that one and not have to redifine a new type?

**Pull request merged**